Add sound support for single media export

drs-11 opened the pull request:
Fixes #1308.

heinezen commented:
@drs-11 Already done? :)

drs-11 commented:
Yeah I thought this was a simple one. I didn't underestimate this did I?

heinezen commented:
@drs-11 Just wanted to make sure that you are done before I review (because you still have the `WIP` tag in the title)

drs-11 commented:
@heinezen the things I have done so far: 
- added wav to opus methods (single file and from drs archive) 
- modified main function to include the added methods
- changed comments and info to describe the sound export methods

If that's all that was needed to be done then please go ahead and review otherwise please mention things I missed.

heinezen commented on openage/convert/tool/singlefile.py:
```diff
@@ -42,11 +42,11 @@ def main(args, error):
-
-    palettes_path = Path(args.palettes_path)
-    palettes = read_palettes(palettes_path)
+    if args.mode not in ["drs-wav", "wav"]:
```
This also needs to check the file extension.

heinezen commented on openage/convert/tool/singlefile.py:
```diff
@@ -42,11 +42,11 @@ def main(args, error):
-    palettes = read_palettes(palettes_path)
+    if args.mode not in ["drs-wav", "wav"]:
+        if not args.palettes_path:
+            raise Exception("palettes-path needs to be specified")
+        palettes_path = Path(args.palettes_path)
```
Newline in between these lines, please

heinezen reviewed (changes requested):
Found one small problem and a formatting error, but looks good otherwise :)

@drs-11 Btw, if you think you are done, then you should request a review in the Github UI (top right) or remove the `[WIP]` label. Then we know that you are ready.

heinezen commented on openage/convert/tool/singlefile.py:
```diff
@@ -42,11 +42,12 @@ def main(args, error):
 
-    if not args.palettes_path:
-        raise Exception("palettes-path needs to be specified")
+    if args.mode not in ["drs-wav", "wav"] or file_extension != "wav":
```
This condition will still be `True` if no `args.mode` was given, but the file extension is `wav`.

Better use `not (args.mode in ["drs-wav", "wav"] or file_extension == "wav")`

drs-11 replied:
ok yeah

heinezen replied:
Boolean logic :D

heinezen reviewed (approved):
Now it works like a charm!
